### ISSUE TITLE
False positive in `extraction_operator_linter()` with plotmath expressions

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#1484]
One can use plotmath to display mathematical expressions in a graphic:

``` r
library(ggplot2)
expr <- list(quote(italic("x")["i"]))
ggplot() + labs(title = parse(text = expr))
```

![](https://i.imgur.com/ATJPjm6.png)

<sup>Created on 2022-07-29 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

But the code for subscript produces an incorrect lint:

``` r
library(lintr)

lint(
  text = "list(quote(italic('x')['i']))",
  linters = extraction_operator_linter()
)
#> <text>:1:23: warning: [extraction_operator_linter] Use `[[` instead of `[` to extract an element.
#> list(quote(italic('x')['i']))
#>                       ^
```

<sup>Created on 2022-07-29 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

The suggestion is misleading as it won't produce the same output:

``` r
library(ggplot2)
expr <- list(quote(italic("x")[["i"]]))
ggplot() + labs(title = parse(text = expr))
```

![](https://i.imgur.com/HyW9E59.png)

<sup>Created on 2022-07-29 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by AshesITR** [comment]
Maybe we could optionally skip `quote()` and friends.

**Comment by IndrajeetPatil** [comment]
Yeah, I agree. We can also skip its close cousins from `{rlang}` package.

**Comment by MichaelChirico** [comment]
I think there are many such cases where `extraction_operator_linter()` is giving false positives (e.g., `[[` method is not defined for the object, or it really is intentional to maintain the class but subset to one element, e.g. `iris["Species"]` is perfectly valid).

Not sure how worth effort it is to maintain exceptions here, authors dinged by this linter often should simply not use it IMO :)

**Comment by MichaelChirico** [comment]
Perhaps a better choice is to add parameters: `allow_single_extraction` and `allow_dollar_extraction` that toggle which of the usages are linted.
